python/ray/data/tests/test_pandas_block.py

```diff
@@ -48,7 +54,203 @@ def fn2(batch):
+        data = {
+            "lists": [
+                [random.randint(0, 100) for _ in range(size)] for _ in range(rows)
+            ],
+        }
+        block = pd.DataFrame(data)
+
+        block_accessor = PandasBlockAccessor.for_block(block)
+        bytes_size = block_accessor.size_bytes()
 
+        true_size = rows * (
+            sys.getsizeof([random.randint(0, 100) for _ in range(size)]) + size * 28
+        )
+        # List overhead + 10 integers per list
+
+        assert true_size * 0.9 <= bytes_size <= true_size * 1.1, (
+            bytes_size,
+            true_size,
+        )
+
+    def test_size_bytes_mixed_types(ray_start_regular_shared):
+        rows = 10_000
+
+        data = {
+            "integers": [random.randint(0, 100) for _ in range(rows)],
+            "floats": [random.random() for _ in range(rows)],
+            "strings": [
+                random.choice(["apple", "banana", "cherry"]) for _ in range(rows)
+            ],
+            "object": [b"\x00" * 128 for _ in range(rows)],
+        }
+        block = pd.DataFrame(data)
+        block_accessor = PandasBlockAccessor.for_block(block)
+        bytes_size = block_accessor.size_bytes()
+
+        # Manually calculate the size
+        int_size = rows * 8
+        float_size = rows * 8
+        str_size = (
+            rows * sum([sys.getsizeof(s) for s in ["apple", "banana", "cherry"]]) // 3
+        )
+        object_size = rows * sys.getsizeof(b"\x00" * 128)
+
+        true_size = int_size + float_size + str_size + object_size
+        assert true_size * 0.9 <= bytes_size <= true_size * 1.1, (bytes_size, true_size)
+
+    def test_size_bytes_nested_lists_strings(ray_start_regular_shared):
+        rows = 5_000
+        size = 10
+        data = {
+            "nested_lists": [
+                [random.choice(["a", "bb", "ccc"]) for _ in range(size)]
+                for _ in range(rows)
+            ],
+        }
+        block = pd.DataFrame(data)
+        block_accessor = PandasBlockAccessor.for_block(block)
+        bytes_size = block_accessor.size_bytes()
+
+        # Manually calculate the size
+        list_overhead = sys.getsizeof(
+            block["nested_lists"].iloc[0]
+        ) + size * sys.getsizeof("bb")
+        true_size = rows * list_overhead
+        assert true_size * 0.9 <= bytes_size <= true_size * 1.1, (bytes_size, true_size)
+
+    def test_size_bytes_multi_level_nesting(ray_start_regular_shared):
+        rows = 1_000
+        size = 10  # if you change this to 1024 it breaks
+        data = {
+            "complex": [
+                {"list": [np.random.rand(size)], "value": {"key": "val"}}
+                for _ in range(rows)
+            ],
+        }
+        block = pd.DataFrame(data)
+        block_accessor = PandasBlockAccessor.for_block(block)
+        bytes_size = block_accessor.size_bytes()
+
+        # Manually calculate the size
+        list_overhead = sys.getsizeof([0] * size) + size * 28
```

Comment:
@Bye-legumes could you take a look at this? if you change size to 1024, it breaks. would be great to get this calculation right

Reply:
I just resolve the calculation of the size which count the string.